internal/access/access.go

```diff
@@ -0,0 +1,42 @@
+package access
+
+import (
+	"context"
+	"github.com/kyma-project/api-gateway/internal/signature"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+func ShouldAllowAccessToV1Beta1(ctx context.Context, k8sClient client.Client) (bool, error) {
+	var cm corev1.ConfigMap
+	err := k8sClient.Get(ctx, client.ObjectKey{Name: "shoot-info", Namespace: "kube-system"}, &cm)
```

Comment:
On k3d without this configmap there is an unnecessary error logged:
```

Warning: Version v1beta1 of APIRule is deprecated and will be removed in future releases. Use version v2 instead.
Error from server (NotFound): error when creating "STDIN": admission webhook "v1beta1-admission.apirule.gateway.kyma-project.io" denied the request: ConfigMap "shoot-info" not found
```
Also, the deprecation message should be an error instead of a warning since v1beta1 is not working anymore.